src/init.cpp

```diff
@@ -908,12 +905,23 @@ bool AppInitParameterInteraction()
 
     // ********************************************************* Step 3: parameter-to-internal-flags
 
-    fDebug = mapMultiArgs.count("-debug");
+    bool fDebug = mapMultiArgs.count("-debug") > 0;
```

Comment:
Now that you've removed the global fDebug bool, you can go ahead and remove it entirely and test the conditions directly (rather than make the variable local), ie:

`if (mapMultiArgs.count("-debug") > 0) {`

and further below:

`if !(GetBoolArg("-nodebug", false) || find(categories.begin(), categories.end(), std::string("0")) != categories.end()) {`